ext/libclementine-tagreader/gmereader.cpp

```diff
@@ -0,0 +1,149 @@
+#include "gmereader.h"
+#include "core/logging.h"
+#include "core/timeconstants.h"
+#include "tagreader.h"
+
+#include <QByteArray>
+#include <QFile>
+#include <QFileInfo>
+#include <QString>
+#include <QtEndian>
+#include <apefile.h>
+#include <tag.h>
+
+bool GME::IsSupportedFormat(const QFileInfo &file_info) {
+  return file_info.completeSuffix().endsWith("spc");
+}
+
+void GME::ReadFile(const QFileInfo &file_info,
+                   pb::tagreader::SongMetadata *song_info) {
+  if (file_info.completeSuffix().endsWith("spc"))
+    GME::SPC::Read(file_info, song_info);
+}
+
+void GME::SPC::Read(const QFileInfo &file_info,
+                    pb::tagreader::SongMetadata *song_info) {
+  QFile file(file_info.filePath());
+  if (!file.open(QIODevice::ReadOnly))
+    return;
+
+  qLog(Debug) << "Reading SPC from file" << file_info.fileName();
+
+  // Check for header -- more reliable than file name alone.
+  if (!file.read(33).startsWith(QString("SNES-SPC700").toLocal8Bit()))
```

Comment:
Unless you really mean `toLocal8Bit()`, you probably want either `toAscii()` or `toUtf8()` everywhere.